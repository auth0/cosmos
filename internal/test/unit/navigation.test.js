import React from 'react'
import { render } from 'enzyme'
import { Icon, Navigation } from '@auth0/cosmos'

describe('Navigation', () => {
  it('renders empty', () => {
    const nav = render(<Navigation description="Test navigation" />)

    expect(nav).toMatchSnapshot()
  })

  it('renders', () => {
    const nav = render(
      <Navigation description="Test navigation">
        <Navigation.Item onClick={() => {}}>
          <Navigation.Item.Text>Dashboard</Navigation.Item.Text>
        </Navigation.Item>
      </Navigation>
    )

    expect(nav).toMatchSnapshot()
  })

  it('renders with icon item', () => {
    const nav = render(
      <Navigation description="Test navigation">
        <Navigation.Item onClick={() => {}}>
          <Icon name="apis" />
          <Navigation.Item.Text>Dashboard</Navigation.Item.Text>
        </Navigation.Item>
      </Navigation>
    )

    expect(nav).toMatchSnapshot()
  })

  it('renders with sub navigation', () => {
    const nav = render(
      <Navigation description="Test navigation">
        <Navigation.Item onClick={() => {}}>
          <Icon name="apis" />
          <Navigation.Item.Text>Dashboard</Navigation.Item.Text>

          <Navigation.Subnav>
            <Navigation.Item>
              <Navigation.Item.Text>Test element</Navigation.Item.Text>
            </Navigation.Item>
          </Navigation.Subnav>
        </Navigation.Item>
        <Navigation.Item onClick={() => {}}>
          <Icon name="apis" />
          <Navigation.Item.Text>Another</Navigation.Item.Text>
        </Navigation.Item>
      </Navigation>
    )

    expect(nav).toMatchSnapshot()
  })

  it('renders navigation groups', () => {
    const nav = render(
      <Navigation description="Test navigation">
        <Navigation.Group title="Group 1">
          <Navigation.Item onClick={() => {}}>
            <Icon name="apis" />
            <Navigation.Item.Text>Dashboard</Navigation.Item.Text>
          </Navigation.Item>
          <Navigation.Item onClick={() => {}}>
            <Icon name="apis" />
            <Navigation.Item.Text>Another</Navigation.Item.Text>
          </Navigation.Item>
        </Navigation.Group>
        <Navigation.Group title="Group 2">
          <Navigation.Item onClick={() => {}}>
            <Icon name="apis" />
            <Navigation.Item.Text>Dashboard</Navigation.Item.Text>
          </Navigation.Item>
          <Navigation.Item onClick={() => {}}>
            <Icon name="apis" />
            <Navigation.Item.Text>Another</Navigation.Item.Text>
          </Navigation.Item>
        </Navigation.Group>
      </Navigation>
    )

    expect(nav).toMatchSnapshot()
  })
})
