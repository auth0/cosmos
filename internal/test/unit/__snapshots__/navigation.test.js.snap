// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders 1`] = `
<styled.nav
  aria-label="Test navigation"
  data-cosmos-key="navigation"
  role="navigation"
>
  <styled.ul
    data-cosmos-key="navigation.group"
    description="Test navigation"
  >
    <NavigationItem
      onClick={[Function]}
      url="#"
    >
      <styled.div>
        Dashboard
      </styled.div>
    </NavigationItem>
  </styled.ul>
</styled.nav>
`;

exports[`Navigation renders empty 1`] = `
<styled.nav
  aria-label="Test navigation"
  data-cosmos-key="navigation"
  role="navigation"
>
  <styled.ul
    data-cosmos-key="navigation.group"
    description="Test navigation"
  />
</styled.nav>
`;

exports[`Navigation renders navigation groups 1`] = `
<styled.nav
  aria-label="Test navigation"
  data-cosmos-key="navigation"
  role="navigation"
>
  <styled.ul
    data-cosmos-key="navigation.group"
    description="Test navigation"
  >
    <Component
      title="Group 1"
    >
      <NavigationItem
        onClick={[Function]}
        url="#"
      >
        <Icon
          color="default"
          name="apis"
          size={20}
        />
        <styled.div>
          Dashboard
        </styled.div>
      </NavigationItem>
      <NavigationItem
        onClick={[Function]}
        url="#"
      >
        <Icon
          color="default"
          name="apis"
          size={20}
        />
        <styled.div>
          Another
        </styled.div>
      </NavigationItem>
    </Component>
    <Component
      title="Group 2"
    >
      <NavigationItem
        onClick={[Function]}
        url="#"
      >
        <Icon
          color="default"
          name="apis"
          size={20}
        />
        <styled.div>
          Dashboard
        </styled.div>
      </NavigationItem>
      <NavigationItem
        onClick={[Function]}
        url="#"
      >
        <Icon
          color="default"
          name="apis"
          size={20}
        />
        <styled.div>
          Another
        </styled.div>
      </NavigationItem>
    </Component>
  </styled.ul>
</styled.nav>
`;

exports[`Navigation renders with icon item 1`] = `
<styled.nav
  aria-label="Test navigation"
  data-cosmos-key="navigation"
  role="navigation"
>
  <styled.ul
    data-cosmos-key="navigation.group"
    description="Test navigation"
  >
    <NavigationItem
      onClick={[Function]}
      url="#"
    >
      <Icon
        color="default"
        name="apis"
        size={20}
      />
      <styled.div>
        Dashboard
      </styled.div>
    </NavigationItem>
  </styled.ul>
</styled.nav>
`;

exports[`Navigation renders with sub navigation 1`] = `
<styled.nav
  aria-label="Test navigation"
  data-cosmos-key="navigation"
  role="navigation"
>
  <styled.ul
    data-cosmos-key="navigation.group"
    description="Test navigation"
  >
    <NavigationItem
      onClick={[Function]}
      url="#"
    >
      <Icon
        color="default"
        name="apis"
        size={20}
      />
      <styled.div>
        Dashboard
      </styled.div>
      <NavigationSubnav>
        <NavigationItem
          url="#"
        >
          <styled.div>
            Test element
          </styled.div>
        </NavigationItem>
      </NavigationSubnav>
    </NavigationItem>
    <NavigationItem
      onClick={[Function]}
      url="#"
    >
      <Icon
        color="default"
        name="apis"
        size={20}
      />
      <styled.div>
        Another
      </styled.div>
    </NavigationItem>
  </styled.ul>
</styled.nav>
`;
